mifosng-android/src/main/java/com/mifos/mifosxdroid/online/clientdetails/ClientDetailsFragment.java

```diff
@@ -530,6 +536,7 @@ public void showClientImageDeletedSuccessfully() {
         Toaster.show(rootView, "Image deleted");
         iv_clientImage.setImageDrawable(ContextCompat.getDrawable(getActivity(), R.drawable
                 .ic_launcher));
+        client_reference.setImagePresent(false);
```

Comment:
Instead of this approach, you should modify the `public void showClientInformation(final Client client) {...}` method by adding another variable `boolean showImageRemoveInMenu` and handle it in that method only.  You don't need to hold the reference of client.